Index: draft.module
===================================================================
RCS file: /cvs/drupal-contrib/contributions/modules/draft/draft.module,v
retrieving revision 1.2.2.41
diff -u -b -r1.2.2.41 draft.module
--- draft.module	15 Mar 2010 21:17:03 -0000	1.2.2.41
+++ draft.module	12 Oct 2010 20:51:22 -0000
@@ -137,7 +139,8 @@
           // This functionality is here specifically for modules that utilize AJAX to create widgets in the form once it
           // Is already loaded in the system so that those fields can be pre-created and then will be populated by the module
           foreach (module_implements('draft') as $module) {
-            module_invoke($module, 'draft', 'form elements', $_GET['draft_id'], unserialize($result->data), $form);
+            //module_invoke($module, 'draft', 'form elements', $_GET['draft_id'], unserialize($result->data), &$form);
+            call_user_func($module.'_draft', 'form elements', $_GET['draft_id'], unserialize($result->data), &$form);
           }
         }
         // button to save the node as a draft for later use
